### PR TITLE
feat(hc): backfill user notification settings missing user_id

### PR DESCRIPTION
**Pull request by vbro**
Duplicate of https://github.com/getsentry/sentry/pull/48707
Closing...

### CODE REVIEW

File: src/sentry/migrations/0439_backfill_notification_user.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ACTOR_TEAM_TYPE = 0
+ACTOR_USER_TYPE = 1
+
+
+def backfill_notification_user(apps, schema_editor):
+    NotificationSetting = apps.get_model("sentry", "NotificationSetting")
+    Actor = apps.get_model("sentry", "Actor")
+    for ns in RangeQuerySetWrapperWithProgressBar(
+        NotificationSetting.objects.filter(user_id=None, team_id=None)
+    ):
```
Comment (vbro) [inline src/sentry/migrations/0439_backfill_notification_user.py:17]: 3,939 records affected

File: src/sentry/migrations/0439_backfill_notification_user.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        try:
+            target = Actor.objects.get(id=ns.target_id)
```
Comment (vbro) [inline src/sentry/migrations/0439_backfill_notification_user.py:19]: select_related not possible without foreign key

File: src/sentry/migrations/0439_backfill_notification_user.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
```
Comment (vbro) [inline src/sentry/migrations/0439_backfill_notification_user.py:38]: Is `is_dangerous = False` okay if only 3,939 records impacted?

Comment (wedamija) [reply src/sentry/migrations/0439_backfill_notification_user.py:38]: Yeah, that's fine.

File: src/sentry/migrations/0439_backfill_notification_user.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/migrations/0439_backfill_notification_user.py:17]: Tbqh I'd skip using the `RangeQuerySetWrapperWithProgressBar` in that case

**Review by wedamija — APPROVED** [review]

**Pull request closed without merging**